File: runtime/flutter/painting/hSVColor.ts
import { IColor } from "../../dart/ui/color";
declare const flutter: {
    painting: {
        hSVColorFromColor: (color: IColor) => IHSVColor;
        hSVColorLerp: (
            a: IHSVColor | undefined,
            b: IHSVColor | undefined,
            t: number
        ) => IHSVColor | undefined;
    };
};
export interface IHSVColor {
    alpha: number;
    hue: number;
    saturation: number;
    value: number;
    withAlpha: (alpha: number) => IHSVColor;
    withHue: (hue: number) => IHSVColor;
    withSaturation: (saturation: number) => IHSVColor;
    withValue: (value: number) => IHSVColor;
    toColor: () => IColor;
    getHashCode: () => number;
    toString: () => string;
}
export class HSVColor {
    public readonly alpha: number = undefined as any;
    public readonly hue: number = undefined as any;
    public readonly saturation: number = undefined as any;
    public readonly value: number = undefined as any;
    public static fromColor(color: IColor): IHSVColor {
        return flutter.painting.hSVColorFromColor(color);
    }
    public static lerp(
        a: IHSVColor | undefined,
        b: IHSVColor | undefined,
        t: number
    ): IHSVColor | undefined {
        return flutter.painting.hSVColorLerp(a, b, t);
    }
    private readonly _dart_withAlpha: (alpha: number) => IHSVColor =
        undefined as any;
    private readonly _dart_withHue: (hue: number) => IHSVColor =
        undefined as any;
    private readonly _dart_withSaturation: (saturation: number) => IHSVColor =
        undefined as any;
    private readonly _dart_withValue: (value: number) => IHSVColor =
        undefined as any;
    private readonly _dart_toColor: () => IColor = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    public withAlpha(alpha: number): IHSVColor {
        return this._dart_withAlpha(alpha);
    }
    public withHue(hue: number): IHSVColor {
        return this._dart_withHue(hue);
    }
    public withSaturation(saturation: number): IHSVColor {
        return this._dart_withSaturation(saturation);
    }
    public withValue(value: number): IHSVColor {
        return this._dart_withValue(value);
    }
    public toColor(): IColor {
        return this._dart_toColor();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
    public toString(): string {
        return this._dart_toString();
    }
}

File: lib/cfr/builtins/libs/flutter/painting/hSVColor.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/painting/colors.dart' as _898f;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedHSVColor extends _36c2.VMManagedBox<_898f.HSVColor> {VMManagedHSVColor({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'alpha'
] = vmObject.alpha;
table  [
'hue'
] = vmObject.hue;
table  [
'saturation'
] = vmObject.saturation;
table  [
'value'
] = vmObject.value;
table  [
'withAlpha'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSVColor>(object: vmObject.withAlpha(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'withHue'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSVColor>(object: vmObject.withHue(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'withSaturation'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSVColor>(object: vmObject.withSaturation(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'withValue'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSVColor>(object: vmObject.withValue(luaCallerArguments  [
1
]?.toDouble()), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'toColor'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_a643.Color>(object: vmObject.toColor(), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(),]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _898f.HSVColor vmObject;

 }

void loadHSVColor({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'hSVColorFromColor'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_898f.HSVColor>(object: _898f.HSVColor.fromColor(_36c2.maybeUnBoxAndBuildArgument<_a643.Color, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'hSVColorLerp'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

final returnValue = _898f.HSVColor.lerp(_36c2.maybeUnBoxAndBuildArgument<_898f.HSVColor?, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_898f.HSVColor?, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState), luaCallerArguments  [
3
]?.toDouble());if(returnValue!= null){return [_36c2.maybeBoxObject<_898f.HSVColor?>(object: returnValue, hydroState: hydroState, table: _36c2.HydroTable()),];}return []; } );
_36c2.registerBoxer<_898f.HSVColor>(boxer: ({required _898f.HSVColor vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedHSVColor(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
